### PR TITLE
DCOS-38853: replace pure-render-mixin with PureComponent

### PR DESCRIPTION
**Pull request by nLight**
as the title suggests. I rewrote the NodesGridView in TS since it was the only component I would have to implement `shouldComponentUpdate` the other way around.

Closes DCOS-38853

## Testing

You may consider smoke testing related pages

## Trade-offs

Not entirely sure with the type definition I wrote for the `NodesGridDials`. `tslint` is fine but vscode tells me I should have a top level `declare`.

### CODE REVIEW

File: plugins/nodes/src/js/components/NodesGridView.tsx [hunk 1]
```diff
@@ -1,49 +1,52 @@
...
+interface NodesGridViewProps {
+  hasLoadingError?: boolean;
+  hiddenServices?: [];
+  hosts: NodesList;
+  receivedEmptyMesosState?: boolean;
+  receivedNodeHealthResponse?: boolean;
+  resourcesByFramework: ResourcesByFramework;
+  selectedResource: string;
+  serviceColors: ServiceColors;
+  services: any[];
```
Comment (brandonc) [inline plugins/nodes/src/js/components/NodesGridView.tsx]: Still new to typescript. Could this be object[]? To at least constrain it that much.

File: plugins/nodes/src/js/components/NodesGridDials.d.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { ClassicComponentClass } from "react";
+
+interface NodesGridDialsProps {
+  hosts: any[];
+  resourcesByFramework: object;
+  selectedResource: string;
+  serviceColors: { [index: string]: number };
+}
+
+const NodesGridDials: ClassicComponentClass<NodesGridDialsProps>;
```
Comment (natmegs) [inline plugins/nodes/src/js/components/NodesGridDials.d.ts]: The error here is just typescript complaining because it requires `declare` for any top level non-interface declarations in .d.ts files. Error goes away with `declare const NodesGridDials: ClassicComponentClass<NodesGridDialsProps>;`

File: plugins/nodes/src/js/components/NodesGridView.tsx [hunk 3]
```diff
@@ -1,49 +1,52 @@
...
-  defaultProps: {
+const MAX_SERVICES_TO_SHOW = 32;
+const OTHER_SERVICES_COLOR = 32;
+
+interface ServiceColors {
+  [index: string]: number;
+}
+
+interface ResourcesByFramework {
+  [index: string]: object;
```
Comment (natmegs) [inline plugins/nodes/src/js/components/NodesGridView.tsx]: It would be cool if we could really take advantage of types here and define what each resource object would look like but that's starting to smell like scope creep....

**Comment by GeorgiSTodorov** [comment]
Not merging this in case @nLight wants to change anything based on the feedback.

**Comment by nLight** [comment]
@brandonc @natmegs thank you very much for the feedback, please give me a harder time 🙇
@GeorgiSTodorov thanks for not rushing it in 💪

**Comment by nLight** [comment]
I introduced a small bug so integration tests were failing. Fixed 🙇
